### ISSUE TITLE
[Feature] Internationalization (i18n) support to the editor

### ISSUE THREAD

**Issue by sharat87** [appsmithorg/appsmith#4404]
Having internationalization (i18n) in the editor canvas can help more diverse teams work on developing applications togather.

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_5w8gc)

**Comment by chen-yaoyu** [comment]
I’m trying to contribute to i18n, but unfortunately, I don’t have the technical resources to change the existing codebase. I found this framework, which I think might help to integrate i18n in the future. I hope this helps & would be very happy to contribute to Chinese localization once the infrastructure is in place. 

https://github.com/formatjs/formatjs

**Comment by Nikhil-Nandagopal** [comment]
Thank you for the wonderful contribution @chen-yaoyu! We'll definitely keep this in mind when we restructure our codebase for i18n :)

**Comment by vmulholl** [comment]
Hello, I'd like to show my interest in having the widgets show other languages as well (French is the most pressing one for myself). 
Ex. the table widget "filter", "download" etc. buttons
For now I can try hiding the bar and creating interactive filters between 2 widgets. 
Let us know if we can help with the translations!

**Comment by shossur** [comment]
We can apply the localization support to the application developed using appsmith right? Also wanted to understand what is the timeline to complete this issue?

**Comment by IAIOMAN117** [comment]
Hi, I also like to show my intereset in having the widgets have option to display in other languages, my case is Spanish, more precisely in Latin American spanish, as the users I have for the app are Mexican and not all of them read english fluently.

Is there any ETA for the feature within the current RoadMap? If needed, I can help providing the necessary text translations if I receive the original texts.

**Comment by okaeiz** [comment]
Is this issue still open? 
How can I contribute to translate and localize Appsmith?

**Comment by ame-appsmith** [comment]
[Another user wanted to have the app translated into Chinese](https://app.intercom.com/a/apps/y10e7138/inbox/inbox/conversation/164629100162044#part_id=comment-164629100162044-16513422459).

**Comment by maiconramos** [comment]
+1

**Comment by thomas-jablonski** [comment]
+1

**Comment by abutalha** [comment]
Please add Arabic and RTL support. we can bring you a huge user base.

**Comment by vishaldfreak** [comment]
We're trying to build ERP for our operations which are handled multiple countries India, China, Dubai, Thailand. It would make the whole app much more user friendly if the UI text, menus, everything could be translated to local language on select of a menu button.

**Comment by talmannn** [comment]
Hi. If would give ability to contribute localisation, I'd help with russian, ukrainian and hebrew. Also, RTL support would be a huge advantage.

**Comment by shossur** [comment]
+1

**Comment by harshaPatilG** [comment]
Hi,
I see, this ticket is opened on May 2021, how early can we expect this feature?

**Comment by shossur** [comment]
I have 1 clarification appsmith is developing this feature for appsmith tool or internationalization support for application developed using appsmith also? If yes how the application developer will be defining the translation text from the appsmith tool?

**Comment by shossur** [comment]
I have 1 clarification appsmith is developing this feature for appsmith tool or internationalization support for application developed using appsmith also? If yes how the application developer will be defining the translation text from the appsmith tool?

**Comment by jansenfavero** [comment]
Olá! desenvolvo soluções web para gestão de informações para empresas e governos locais aqui no Brasil, gostaria muito de poder **personalizar os textos de datas, filtros e campo de busca**, no AppSmith

**Comment by IAIOMAN117** [comment]
Almost a year since I started to try to move the applicaiton I made from PoC to MVP, but sadly my users speak spanish only and the usability for them is very limited.
I understand that this might not be on your roadmap, but there must be a way to at least workaround it. And I mean for the text that are inside the controls, If this could be added at least to the settings of each control I can deal with having to manually override each and single of them, but not having the posibility is a deal breaker for me.

**Comment by Nikhil-Nandagopal** [comment]
@vidz1979 so are you suggesting that right now you just want a way to specify the language so that chrome does not pop-up with this annoying notification?
I agree with your approach here and we do plan on introducing language packs for all our widgets in the future but as you said that is a hard thing to achieve and we want to get it right.

**Comment by gehtalles-de** [comment]
@Nikhil-Nandagopal Is this topic anywhere on the actual development roadmap? Native language management would be a very important feature for me (and I guess many others from non-english speaking countries) to use Appsmith for business use cases..

**Comment by vidz1979** [comment]
@gehtalles-de totally agree. It's a must have.

**Comment by guntribam** [comment]
Please put this in you priority backlog

**Comment by ayeah** [comment]
> [Another user wanted to have the app translated into Chinese](https://app.intercom.com/a/apps/y10e7138/inbox/inbox/conversation/164629100162044#part_id=comment-164629100162044-16513422459).

A chinese version has been released, called pageplug, but it uses old version of appsmith, I still want official i18n support, please add this feature, allowing cummutity members to translate and share language pack files, including editor and apps.

**Comment by victorcai2006** [comment]
> > [Another user wanted to have the app translated into Chinese](https://app.intercom.com/a/apps/y10e7138/inbox/inbox/conversation/164629100162044#part_id=comment-164629100162044-16513422459).
> 
> A chinese version has been released, called pageplug, but it uses old version of appsmith, I still want official i18n support, please add this feature, allowing cummutity members to translate and share language pack files, including editor and apps.
Yes, I also found that the version of Pageplug is older and there are some bugs that have not been fixed, so I just use AppSmith. If I can, I can also provide my help for the sinicization of Chinese.

**Comment by Skarema** [comment]
Please implement i18n functionality for the editor and the system itself.

"Select Date" placeholder is hardcoded... The text inside the file picker is hardcoded... 

Using the browser translation feature is an option, but having to explain this even to internal users is a pain.

**Comment by aeturco** [comment]
about Datepicker, for me it is a big issue for the customer to see the english words instead of the italian ones. Please I really need the Datepicker in Italian. Thanks.

**Comment by pongraczi** [comment]
I need to support 3 languages.  
Creating three identical pages with different language seems unpractical.

**Comment by rahulbarwal** [comment]
@pongraczi Can you please share more about your case? 
Could it be solved by something like this: https://community.appsmith.com/template/multilingual-app-ai-translations ?

Let me know if you have any more questions about this template.

**Comment by pongraczi** [comment]
@rahulbarwal Thank you for your quick response!

The application could show the page in 3 languages: HU, EN, DE  
Technically it is just a form, but the translation must be exactly as the user defined, so, automatic translation is not really useful in my case.  
I need exact words/sentences.

Otherwise the multilingual app ai translation seems ok, but not in my case. I can imagine that kind of ai translation for a blog or something less restrictive contents, could be far better than a google translate (especially from/to HU, sometimes it is still messy). 

Thanks, I check the app to learn something new :)

**Comment by rahulbarwal** [comment]
@pongraczi If you require the exact words and sentences, I don't believe AI is the best solution for this. You might consider hardcoding that in your JavaScript object or fetching it from your backend APIs. However, the core logic for updating the languages in the AppSmith UI will remain the same as outlined in the template above. 
Please feel free to loop me in if you have any questions regarding this.

**Comment by sozenstrapi** [comment]
Language plugins need to be made. At least the language file should be exported, and if necessary, full translations should be possible with the help of the community.

**Comment by pongraczi** [comment]
Technically I solved the core problem with the i18next library.  
Some internal widgets are still in English, but I can live with it :)

**Comment by rahulbarwal** [comment]
As a feature this is not a priority.

@sozenstrapi Up to an extent translations are possible with what @pongraczi is saying.  
Are there any challenges that you are facing with that?  Please share, we would love to help.  

And also if there are any internal widgets which you want fixed on priority, let us know, we'll see if we can prioritize them.

**Comment by StoneMoshe** [comment]
We are evaluating Appsmith for a kiosk management SaaS project that requires complete RTL (Hebrew/Arabic) support.

Without RTL, aligning labels, input fields, and layouts for Hebrew/Arabic users becomes very challenging and severely impacts usability.

This feature is critical for our use case, and we kindly ask that you consider prioritizing it. If a beta program for RTL becomes available, we would be happy to participate and provide feedback.

**Comment by kboykov** [comment]
I also think that translations is a critical feature. I'm not even talking about full app translation support. Only errors/validation/system notifications/default texts etc is enough as a start.

**Comment by automatron3k** [comment]
This issue has already more than 4 years, I do not either have the techinal skills to edit the codebase, but it would be really helpful to allow developers modify placerholders and other elements directly in the GUI editor.

**Comment by mickaelboudreau** [comment]
As a francophone organization, we need proper localization for certain widgets — especially the DatePicker and FilePicker. While labels can be customized, the built-in UI elements (month names, days, system text, etc.) remain in English.

Having full translation support or the ability to override these widget texts is important for providing a consistent experience in French. We would appreciate native localization options or any updates regarding internationalization for these components.

Thanks for considering this improvement.

**Comment by 999ml** [comment]
**https://github.com/999ml/appsmith-i18n**
